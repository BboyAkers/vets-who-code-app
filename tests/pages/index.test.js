import React from 'react';
import IndexPaege from '../../src/pages/index';

describe('<IndexPaege />', () => {
  let wrapper;


  beforeEach(() => wrapper = shallow(<IndexPaege />));

  test('should update email property in state onChange', () => {
    const expected = 'someuser@mail.com';
    const mockEvent = {
      target: {
        name: 'email',
        value: expected
      }
    };
    wrapper.instance().handleInputChange(mockEvent);
    expect(wrapper.state('email')).toEqual(expected);
  });

  test('should call window.fetch when handleUserSubscribe is invoked', () => {
    const mockEvent = {
      preventDefault: jest.fn()
    };

    window.fetch = jest.fn().mockImplementation(() => Promise.resolve());

    wrapper.instance().handleUserSubscribe(mockEvent);
    expect(window.fetch).toHaveBeenCalled();
  });


  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
