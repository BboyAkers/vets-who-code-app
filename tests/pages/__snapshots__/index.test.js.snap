// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndexPaege /> should render correctly 1`] = `
<TemplateWrapper>
  <Header />
  <section
    className="section pad-regular bg-default our_stories small-top-pad"
    id="our_stories"
  >
    <div
      className="container"
    >
      <div
        className="row bg-dark"
      >
        <div
          className="col-md-5 col-sm-12 no_left_pad no_right_pad"
        >
          <img
            alt="Vets Who Code veterans at Google event."
            className="img-responsive"
            src="image-mock"
          />
        </div>
        <div
          className="col-md-7 col-sm-12 our_story_content text-center"
        >
          <div
            className="featured-heading"
          >
            <h2>
              #VetsWhoCode
            </h2>
          </div>
          <p>
            #VetsWhoCode is a veteran-led and operated 501(c)(3) charitable non-profit that focuses on training veterans in web development and software engineering principles free of charge with the focus of starting careers as javascript developers.
          </p>
        </div>
      </div>
    </div>
  </section>
  <section
    className="section pad-regular bg-light our_missions"
    id="our_missions"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="featured-heading text-center"
        >
          <h2
            className="dark_color"
          >
            RETOOL
            <i
              aria-hidden="true"
              className="fa fa-code"
            />
             RETRAIN
            <i
              aria-hidden="true"
              className="fa fa-code"
            />
             RELAUNCH
          </h2>
        </div>
        <div
          className="col-sm-12 our_mission_content text-center"
        >
          <blockquote>
            <p>
              Through training, building projects, mentorship, and opportunities to connect with industry experts, we ensure that our troops gain the best knowledge possible in web development and software engineering.
            </p>
          </blockquote>
        </div>
      </div>
    </div>
  </section>
  <section
    className="section bg-default causes-card"
    id="causes"
    style={
      Object {
        "padding": "75px 0",
      }
    }
  >
    <div
      className="container"
    >
      <div
        className="normal_heading text-center"
      >
        <h2>
          Our Partners
        </h2>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-12 cause_content text-center"
          style={
            Object {
              "marginBottom": 40,
            }
          }
        >
          <h4
            className="cause_title"
          >
            Thank You For Working With #VetsWhoCode!
          </h4>
          <hr />
          <h3
            className="subtitle"
          >
            We Are Grateful To Have These Companies Partner With Us On Our Journey To Train Veterans How To Code!.
          </h3>
        </div>
      </div>
      <SponsorSlider />
    </div>
  </section>
  <section
    className="section bg-dark pad-regular event_card"
    id="event_card"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-6 event_content"
        >
          <h4>
            Next Cohort
          </h4>
          <div
            className="event_excerpt"
          >
            <p>
              Countdown until our next #VetsWhoCode Cohort gets started:
            </p>
          </div>
        </div>
        <Countdown />
      </div>
    </div>
  </section>
  <section
    className="section bg-light pad-regular newsletter_card"
    id="newsletter_card"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12 newsletter_wrapper"
        >
          <div
            className="newsletter_inner_wrapper"
          >
            <div
              className="row"
            >
              <div
                className="col-md-4"
              >
                <h3
                  className="text-center"
                >
                  JOIN OUR EMAIL LIST
                </h3>
              </div>
              <div
                className="col-md-8"
              >
                <form
                  id="s2do-form"
                  onSubmit={[Function]}
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-md-9"
                    >
                      <div
                        className="form-group"
                      >
                        <input
                          className="form-control"
                          id="subscriber-email"
                          name="email"
                          onChange={[Function]}
                          placeholder="Enter your email address"
                          required={true}
                          type="email"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      className="col-md-3 subscribe-button_box"
                    >
                      <button
                        className="btn btn-subscribe"
                        id="subscribe-button"
                        type="submit"
                      >
                        subscribe
                      </button>
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</TemplateWrapper>
`;
